File: 5 kyu/Pick peaks/pick_peaks.py
y = [1, 2, 3, 4, 4, 4, 5, 4, 3, 2, 3, 4, 5, 4]



def pick_peaks(arr):
    pos = []
    prob_peak = False
    for i in range(1, len(arr)):
        if arr[i] > arr[i-1]:
            prob_peak = i
            print(arr[i])
        elif arr[i] < arr[i-1] and prob_peak:
            pos.append(prob_peak)
            prob_peak = False
    return {'pos':pos, 'peaks':[arr[i] for i in pos]}


test = pick_peaks(y)

print(test)

File: 5 kyu/Pick peaks/pick_peaks.js



function pickPeaks(arr) {
    return arr.reduce(
        (prev, cur, index) => {
            if (cur > arr[index - 1] && cur > arr.slice(index).find(el => el !== cur)) {
                prev.pos.push(index), prev.peaks.push(cur);
            }
            return prev;
        },
        { pos: [], peaks: [] }
    );
}


console.log(pickPeaks([1,2,3,4,4,4,3,3,2,5,2]))
